cdci_data_analysis/flask_app/dispatcher_query.py

```diff
@@ -411,6 +411,35 @@ def free_up_space(app):
 
         return jsonify(dict(output_status=result_scratch_dir_deletion))
 
+    @staticmethod
+    def get_user_specific_instrument_list(app):
+        token = request.args.get('token', None)
+
+        roles = None
+        if token is not None:
+            app_config = app.config.get('conf')
+            secret_key = app_config.secret_key
+            output, output_code = tokenHelper.validate_token_from_request(token=token, secret_key=secret_key,
+                                                                          action="getting the list of instrument")
+            decoded_token = tokenHelper.get_decoded_token(token, secret_key)
+            roles = tokenHelper.get_token_roles(decoded_token)
+            if output_code is not None:
+                return make_response(output, output_code)
+
+        out_instrument_list = []
+        for instrument_factory in importer.instrument_factory_list:
+            instrument = instrument_factory()
+
+            if instrument.development:
```

Comment:
this logic can all go in the method `check_access`

Reply:
now inside the `check_instrument_access`